### PR TITLE
Add language support for sway

### PR DESCRIPTION
**Pull request by sdankel**
Adds language support for the [sway](https://github.com/FuelLabs/sway) smart contract programming language using [tree-sitter-sway](https://github.com/FuelLabs/tree-sitter-sway/).

### CODE REVIEW

**Comment by sdankel** [comment]
I found a few bugs in the tree sitter implementation. Will move out of draft once those are fixed.

**Comment by kirawi** [comment]
It looks like `highlights.scm` is a copy of the Rust version. You can replace it all with `; inherits: rust`

**Comment by sdankel** [comment]
> It looks like `highlights.scm` is a copy of the Rust version. You can replace it all with `; inherits: rust`

Good to know!

I've updated it to include the nuances for `sway` in the `tree-sitter` repo here: https://github.com/FuelLabs/tree-sitter-sway/blob/master/queries/highlights.scm

Does helix use the tree-sitter `queries` directory by default, or do I need to include the same `queries` directory in helix?

File: book/src/generated/lang-support.md [hunk 1]
```diff
@@ -124,6 +124,7 @@
 | sshclientconfig | ✓ |  |  |  |
 | starlark | ✓ | ✓ |  |  |
 | svelte | ✓ |  |  | `svelteserver` |
+| sway | ✓ | ✓ | ✓ | `sway-lsp` |
```
Comment (sdankel) [inline book/src/generated/lang-support.md:127]:
```suggestion
| sway | ✓ | ✓ | ✓ | `forc lsp` |
```

File: languages.toml [hunk 2]
```diff
@@ -52,6 +52,20 @@ args = { attachCommands = [ "platform select remote-gdb-server", "platform conne
 name = "rust"
 source = { git = "https://github.com/tree-sitter/tree-sitter-rust", rev = "0431a2c60828731f27491ee9fdefe25e250ce9c9" }
 
+[[language]]
+name = "sway"
+scope = "source.sway"
+injection-regex = "sway"
+file-types = ["sw", ".sw"]
```
Comment (the-mikedavis) [inline languages.toml:59]:
```suggestion
file-types = ["sw"]
```

This file-type will match `*.sw` files, no need for an extra `.sw` one

**Review by the-mikedavis — APPROVED** [review]

**Pull request merged**